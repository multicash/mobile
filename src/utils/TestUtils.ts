import fs from 'fs'
import path from 'path'
import { compileVueToRn } from 'vue-native-scripts'

export const getTestComponent = (component: string): any => {
  const data = fs.readFileSync(path.join(__dirname, '..', component), 'utf8')

  return compileVueToRn(data).output
}
