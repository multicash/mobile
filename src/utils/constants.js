export default {
  priceApi: 'https://api.vergecurrency.network/price/api/v1/price',
  ipApi: 'https://api.vergecurrency.network/price/api/v1/ip/',
  vwsApi: 'https://api.vergecurrency.network/vws/api',
  bnApi: 'https://api.vergecurrency.network/node/api/XVG',
  explorer: 'https://verge-blockchain.info',
  termsOfUse: 'https://vergecurrency.com/wallets/terms',
  feePerKb: 100000,
  satoshiDivider: 1000000,
  decimalPerSatoshi: 6,
  defaultCurrencyCode: 'USD',
  paperKeyLength: 12
}
