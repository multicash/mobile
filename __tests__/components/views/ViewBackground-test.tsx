import React from 'react'
import renderer from 'react-test-renderer'
import { getTestComponent } from '@/utils/TestUtils'

const ViewBackground = getTestComponent('components/views/ViewBackground.vue')

describe('Render view background component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ViewBackground />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
