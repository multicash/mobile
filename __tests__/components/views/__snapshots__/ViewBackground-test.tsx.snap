// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render view background component renders correctly 1`] = `
<import __react__vue__Vue, { observer as __react__vue__observer } from 'vue-native-core'
import __react__vue__ReactNative from 'react-native'
import __react__vue__PropType from 'prop-types'
import { buildNativeComponent as __react__vue__buildNativeComponent } from 'vue-native-helper'
import { bindNativeClass as __react__vue__bindClass,bindNativeStyle as __react__vue__bindStyle,mergeNativeStyleAndNativeClass as __react__vue__mergeNativeStyleAndNativeClass,mergeProps as __react__vue__mergeProps,renderSlot as __react__vue__renderSlot } from 'vue-native-helper'
import {
      createElement as __react__vue__createElement,
      Component as __react__vue__Component
    } from 'react'


const __react__vue__options =  {
  name: 'ViewBackground',

  props: {
    noPadding: {
      type: Boolean,
      default: false
    },

    scrollable: {
      type: Boolean,
      default: false
    }
  },

  computed: {
    styles () {
      return stylesStore(this.isDarkScheme, this.noPadding, this.scrollable)
    }
  },

  methods: {
    scrollTo (options) {
      if (this.$refs.scrollView) {
        this.$refs.scrollView.scrollTo(options)
      }
    }
  }
}

const stylesStore = (isDarkScheme, noPadding, scrollable) => {
  return {
    safeArea: {
      flex: 1,
      backgroundColor: isDarkScheme ? '#222429' : '#ededf3'
    },

    viewBackground: {
      flex: 1,
      padding: scrollable ? undefined : (noPadding ? 0 : 20),
      flexDirection: 'column',
      backgroundColor: isDarkScheme ? '#222429' : '#ededf3'
    },

    contentContainer: {
      padding: noPadding ? 0 : 20
    }
  }
}

const __react__vue__render = function render(vm) {
    const __react__vue__slotSet = __react__vue__renderSlot.call(this, [], this.props.children);
    return vm['scrollable'] ? __react__vue__createElement(vm.$options.components['ScrollView'], __react__vue__mergeProps.call(this, this.props.__react__vue__nativeEvents, {
        ref: ref => {
            this.setRef(ref, "scrollView", false);
            this.setRootRef(ref);
            this.props['__react__vue__setRef'] && this.props['__react__vue__setRef'](ref);
        },
        contentContainerStyle: vm['styles'].contentContainer,
        contentInsetAdjustmentBehavior: "automatic",
        disableScrollViewPanResponder: true,
        style: __react__vue__mergeNativeStyleAndNativeClass(__react__vue__bindClass.call(this, {
            parentClass: this.props.style
        }), __react__vue__bindStyle(vm['styles'].viewBackground, undefined, undefined))
    }), __react__vue__slotSet(undefined, {})) : __react__vue__createElement(vm.$options.components['View'], __react__vue__mergeProps.call(this, this.props.__react__vue__nativeEvents, {
        ref: ref => {
            this.setRootRef(ref);
            this.props['__react__vue__setRef'] && this.props['__react__vue__setRef'](ref);
        },
        style: __react__vue__mergeNativeStyleAndNativeClass(__react__vue__bindClass.call(this, {
            parentClass: this.props.style
        }), __react__vue__bindStyle(vm['styles'].viewBackground, undefined, undefined))
    }), __react__vue__slotSet(undefined, {}));
};

const __react__vue__css = {}

const __react__vue__ComponentBuilded = __react__vue__buildNativeComponent(__react__vue__render, __react__vue__options, {Component: __react__vue__Component, PropTypes: __react__vue__PropType, Vue: __react__vue__Vue, ReactNative: __react__vue__ReactNative, css: __react__vue__css})

export default __react__vue__observer(__react__vue__ComponentBuilded) />
`;
